Makefile

```diff
@@ -77,16 +85,42 @@ test:
 doc:
 	$(CARGO) doc
 
+e2e: build
+	(cd ./api_tests; yarn install; yarn test)
+
 check_format: check_rust_format check_toml_format check_ts_format
 
+STAGED_FILES = $(shell git diff --staged --name-only)
```

Comment:
📣 you have changed the behaviour and considering how I use git/CLion I disagree with the usage of `--staged` and would prefer to keep the original behaviour `git status --untracked-files=no --short` simply because I/CLion don't stage changes before a commit meaning I am going to have a lot of CI fails due to formatting.

Reply:
I don't quite understand what you mean. You cannot commit files without staging them and AFAIK CLion actually does the opposite of what you are saying - it usually stages files automatically.

I don't see how the new behaviour is bad. On the contrary, it is actually more correct because it only runs the `check_XY_format` tasks for the files that are about to be committed. If you have changes in TS files but you are not committing them, the Makefile does no longer care.

Reply:
ping @D4nte

Reply:
ok. let me test.

Reply:
ok, looks like my memory is playing with me.